File: translated/f-to-c/DRB069-sectionslock1-orig-no.c
/*
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Copyright (c) 2017-20, Lawrence Livermore National Security, LLC
and DataRaceBench project contributors. See the DataRaceBench/COPYRIGHT file for details.

SPDX-License-Identifier: (BSD-3-Clause)
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
*/

//Two tasks with a lock synchronization to ensure execution order. No data race pairs.

#include <omp.h>
#include <stdio.h>

int main()
{
    omp_lock_t lock;
    int i;

    i = 0;
    omp_init_lock(&lock);

    #pragma omp parallel sections
    {
        #pragma omp section
        {
            omp_set_lock(&lock);
            i = i + 1;
            omp_unset_lock(&lock);
        }
        #pragma omp section
        {
            omp_set_lock(&lock);
            i = i + 2;
            omp_unset_lock(&lock);
        }
    }

    omp_destroy_lock(&lock);

    printf("I = %d\n", i);

    return 0;
} 

File: translated/f-to-go/DRB069-sectionslock1-orig-no.go
//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
// Copyright (c) 2017-20, Lawrence Livermore National Security, LLC
// and DataRaceBench project contributors. See the DataRaceBench/COPYRIGHT file for details.
//
// SPDX-License-Identifier: (BSD-3-Clause)
//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

//Two tasks with a lock synchronization to ensure execution order. No data race pairs.

package main

import (
	"fmt"
	"sync"
)

func main() {
	var lock sync.Mutex
	var i int

	i = 0

	//$omp parallel sections
	var wg sync.WaitGroup

	// Section 1
	wg.Add(1)
	go func() {
		defer wg.Done()
		lock.Lock() // omp_set_lock equivalent
		i = i + 1
		lock.Unlock() // omp_unset_lock equivalent
	}()

	// Section 2
	wg.Add(1)
	go func() {
		defer wg.Done()
		lock.Lock() // omp_set_lock equivalent
		i = i + 2
		lock.Unlock() // omp_unset_lock equivalent
	}()

	wg.Wait()
	//$omp end parallel sections

	// omp_destroy_lock equivalent - handled automatically by Go

	fmt.Printf("I = %3d\n", i)
}
